exporter/collector/config.go

```diff
@@ -275,7 +285,17 @@ func generateClientOptions(ctx context.Context, clientCfg *ClientConfig, cfg *Co
 			copts = append(copts, option.WithEndpoint(clientCfg.Endpoint))
 		}
 	}
-	if cfg.ImpersonateConfig.TargetPrincipal != "" {
+	// If the AccessTokenPath environment variable is defined, then use that
+	// for credentials.
+	accessTokenFilePath := os.Getenv(AccessTokenPath)
+	if accessTokenFilePath != "" {
+		fts := fileTokenSource{}
+		if _, err := fts.Token(); err != nil {
+			return nil, fmt.Errorf("parsing access token file: %v", err)
+		}
+		tokenSource := oauth2.ReuseTokenSource(nil, fts)
```

Comment:
Should we consider using `ReuseTokenSourceWithExpiry`, and make `earlyExpiry` configurable via an environment variable (default to 0)?

```go
func ReuseTokenSourceWithExpiry(t *Token, src TokenSource, earlyExpiry time.Duration) TokenSource
```

Given that we're going to use this to export logs, it would be nice if we refreshed the credentials earlier to avoid more logs getting buffered.

Reply:
yes good call, done.